app-android/src/main/java/com/ustadmobile/port/android/view/CourseDiscussionDetailFragment.kt

```diff
@@ -145,6 +185,124 @@ class CourseDiscussionDetailFragment: UstadDetailFragment<CourseDiscussion>(),
             ustadFragmentTitle = value?.courseDiscussionTitle
         }
 
+}
 
 
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun CourseDiscussionDetailScreen(
+    uiState: CourseDiscussionDetailUiState = CourseDiscussionDetailUiState(),
+    onClickAddPost: () -> Unit = {},
+    onClickPost: (DiscussionPostWithDetails) -> Unit = {},
+    onClickDeletePost: (DiscussionPostWithDetails) -> Unit = {},
+){
+
+    LazyColumn{
+
+        //Description:
+        item{
+
+            Text(
+                uiState.courseDiscussion?.courseDiscussionDesc?:"",
+                style = Typography.body1,
+                modifier = Modifier.padding(8.dp)
+            )
+        }
+
+        item{
+            Spacer(modifier = Modifier.height(10.dp))
+        }
+
+        item{
+            Text(stringResource(R.string.posts),
+                style = Typography.h4,
+                modifier = Modifier.padding(8.dp))
+        }
+
+
+        items(
+            items = uiState.posts,
+            key = { post -> post.discussionPostUid }
+        ){ post ->
+
+            val context = LocalContext.current
+            val datePosted = remember { DateFormat.getDateFormat(context)
```

Comment:
Please use rememberFormattedDate. This doesn't have a key set, so we haven't told compose what would invalidate it. There needs to be an explicit invalidation key, e.g. the timestamp. This is done by the rememberformatteddate function.